examples/demo/modules/blog/cms_blog_api.e

```diff
@@ -8,93 +8,108 @@ class
 	CMS_BLOG_API
 
 inherit
-	CMS_NODE_API
+	CMS_MODULE_API
+		rename
+			make as make_with_cms_api
 		redefine
-			initialize,
-			node_storage
+			initialize
 		end
 
+	REFACTORING_HELPER
+
 create
 	make
 
-feature {NONE} -- Implementation
+feature {NONE} -- Initialization
+
+	make (a_api: CMS_API; a_node_api: CMS_NODE_API)
+			-- (from CMS_MODULE_API)
+			-- (export status {NONE})
+		do
+			node_api := a_node_api
+			make_with_cms_api (a_api)
+		end
 
 	initialize
 			-- <Precursor>
 		do
 			Precursor
-			-- Create the node storage for type blog
+
+				-- Create the node storage for type blog
 			if attached {CMS_STORAGE_SQL_I} storage as l_storage_sql then
-				create {CMS_BLOG_STORAGE_SQL} node_storage.make (l_storage_sql)
+				create {CMS_BLOG_STORAGE_SQL} blog_storage.make (l_storage_sql)
 			else
-				create {CMS_BLOG_STORAGE_NULL} node_storage.make
+				create {CMS_BLOG_STORAGE_NULL} blog_storage.make
 			end
-			initialize_node_types
+--			initialize_node_types
 		end
 
+feature {CMS_API_ACCESS, CMS_MODULE, CMS_API} -- Restricted access		
+
+	node_api: CMS_NODE_API
+
 feature {CMS_MODULE} -- Access nodes storage.
 
-	node_storage: CMS_BLOG_STORAGE_I
+	blog_storage: CMS_BLOG_STORAGE_I
+
+feature -- Configuration of blog handlers
+
+	entries_per_page : NATURAL_32 = 2
```

Comment:
at some point we will need to be able to change this settings, so it is better in the BLOG API itself, rather than using an interface by inheritance.